server/reload.go

```diff
@@ -115,41 +233,86 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 		}
 		switch strings.ToLower(field.Name) {
```

Comment:
I'm not sure how to decompose this `switch` into multiple methods cleanly. If we had `handleTraceOptions`, `handleTLSOptions`, etc., we would still have to switch to determine which handler to call.